build.sc

```diff
@@ -33,7 +33,7 @@ import de.tobiasroeser.mill.vcs.version.VcsVersion
 
 val scala213 = "2.13.10"
 val scala212 = "2.12.17"
-val scala3 = "3.2.2"
+val scala3 = "3.3.3"
```

Comment:
Not sure why this update is exactly needed.

I ran into a seemingly unrelated error:

```
[error] error while loading TimeUnit,
[error] class file /modules/java.base/java/util/concurrent/TimeUnit.class is broken, reading aborted with class java.lang.RuntimeException
[error] bad constant pool index: 0 at pos: 3355
```

which is likely due to me building on a machine with JDK >= 21.

In any case, I think we can pumb this since Scala 3.3.x is an LTS anyway.